kuma/javascript/src/task-completion-survey.test.js

```diff
@@ -0,0 +1,222 @@
+        expect(mockGA1.mock.calls[0]).toEqual(['set', 'dimension14', 'Yes']);
+        expect(mockGA1.mock.calls[1][0]).toBe('send');
+        expect(mockGA1.mock.calls[1][1].hitType).toBe('event');
+        expect(mockGA1.mock.calls[1][1].eventCategory).toBe('survey');
+        expect(mockGA1.mock.calls[1][1].eventAction).toBe('prompt');
+        expect(mockGA1.mock.calls[1][1].eventValue).toBe('impression');
+    });
+
+    test('does not render if waffle flag is not set', () => {
+        mockUserData.waffle.flags[WAFFLE_FLAG] = false;
+        localStorage.clear();
+
+        const tcs = create(
+            <GAProvider.context.Provider value={mockGA2}>
+                <LocaleProvider.context.Provider value={mockLocale}>
+                    <DocumentProvider.context.Provider value={mockDocumentData}>
+                        <UserProvider.context.Provider value={mockUserData}>
+                            <TaskCompletionSurvey />
+                        </UserProvider.context.Provider>
+                    </DocumentProvider.context.Provider>
+                </LocaleProvider.context.Provider>
+            </GAProvider.context.Provider>
+        );
+
+        // It should not render anything
+        expect(tcs.toJSON()).toBe(null);
+    });
+
+    test('does not render if embargoed', () => {
+        mockUserData.waffle.flags[WAFFLE_FLAG] = true;
+        localStorage.setItem('taskTracker', String(Date.now() + 100000));
+
+        const tcs = create(
+            <GAProvider.context.Provider value={mockGA2}>
+                <LocaleProvider.context.Provider value={mockLocale}>
+                    <DocumentProvider.context.Provider value={mockDocumentData}>
+                        <UserProvider.context.Provider value={mockUserData}>
+                            <TaskCompletionSurvey />
+                        </UserProvider.context.Provider>
+                    </DocumentProvider.context.Provider>
+                </LocaleProvider.context.Provider>
+            </GAProvider.context.Provider>
+        );
+
+        // It should not render anything
+        expect(tcs.toJSON()).toBe(null);
+    });
+
+    test('does render if embargo has expired', () => {
+        mockUserData.waffle.flags[WAFFLE_FLAG] = true;
+        localStorage.setItem('taskTracker', String(Date.now() - 1000));
+
+        const tcs = create(
+            <GAProvider.context.Provider value={mockGA2}>
+                <LocaleProvider.context.Provider value={mockLocale}>
+                    <DocumentProvider.context.Provider value={mockDocumentData}>
+                        <UserProvider.context.Provider value={mockUserData}>
+                            <TaskCompletionSurvey />
+                        </UserProvider.context.Provider>
+                    </DocumentProvider.context.Provider>
+                </LocaleProvider.context.Provider>
+            </GAProvider.context.Provider>
+        );
+
+        // It should not render anything
```

Comment:
I believe this comment is incorrect here?

Reply:
Oops. I'll fix that.

Reply:
Fixed